### PR TITLE
Auth and Cart Setup

### PR DESCRIPTION
**Pull request by srishti-maurya**
## Added 
- Authentication (Login and Logout)
- Add to cart feature

### CODE REVIEW

File: src/frontend/contexts/auth-context.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    try {
+      const { data } = await loginRequest(loginInput);
+      localStorage.setItem("token", JSON.stringify(data.encodedToken));
+      localStorage.setItem("isLoggedIn", JSON.stringify(true));
+      setToken(data.encodedToken);
+      setLoginInput({ email: "", password: "" });
+      navigate("/");
+      setIsLoggedIn(true);
+    } catch (error) {
+      console.log(error);
```
Comment (pritam-kr) [inline src/frontend/contexts/auth-context.js:38]: Suggestion:  removed error handling with console

File: src/frontend/pages/Home/components/SectionCollection.jsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    return (
+      <div
+        className="category-wrapper"
+        onClick={() => {
+          dispatch({ type: changeType });
+          navigate("/products");
+        }}
+      >
+        <div className="category-body">
+          <img src={image} alt="image" className="img-responsive" />
```
Comment (pritam-kr) [inline src/frontend/pages/Home/components/SectionCollection.jsx:18]: suggestion: you can use the product title in alt attribute

**Comment by pritam-kr** [comment]
Suggestion: 
- Removed wishlist and cart counting as a zero in the initial rendering.
- And you can use toast to notify users.

**Review by pritam-kr — APPROVED** [review]

File: src/App.js [hunk 3]
```diff
@@ -11,6 +10,9 @@ function App() {
         <Route path="/" element={<Home />} />
         <Route path="/mock" element={<MockAPI />} />
         <Route path="/products" element={<Products />} />
+        <Route path="/cart" element={<Cart />} />
+        <Route path="/login" element={<Login />} />
```
Comment (himadri2110) [inline src/App.js:14]: The cart should be a private route. Only the logged in users should be able to access it

File: src/frontend/components/Card.jsx [hunk 4]
```diff
@@ -45,7 +48,7 @@ export function Card({
                 </div>
               </div>
             </div>
-            <button className="btn btn-sm">
+            <button className="btn btn-sm" onClick={() => addToCart(item)}>
```
Comment (himadri2110) [inline src/frontend/components/Card.jsx:51]: If the item is added to the cart, the button should show 'Go to Cart' and redirect to Cart on clicking it.

File: src/frontend/contexts/cart-context.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+export const useCart = () => useContext(CartContext);
+
+export function CartProvider({ children }) {
+  const { token, isLoggedIn } = useAuth();
+  const [cartItems, setCartItems] = useState([]);
+
+  function addToCart(product) {
+    if (isLoggedIn) {
+      (async function () {
+        try {
```
Comment (himadri2110) [inline src/frontend/contexts/cart-context.js:16]: If the user is not logged in, you can redirect to the login page.

**Review by himadri2110 — APPROVED** [review]
The home page is nicely divided into separate reusable components. Good work.

**Pull request merged**